Brief-over-of-Heads-and-firmware-security/FAQ.md

```diff
@@ -0,0 +1,139 @@
+ represents increased "surface area" for attacks, as well as unnecessary
+ complexity in the boot process.
+
+coreboot is open source and focuses on just the code necessary to bring the
+ system up from reset. This minimal code base has a much smaller surface area
+ and is possible to audit. Additionally, self-help is possible if custom
+ features are required or if a security vulnerability needs to be patched.
+
+What's wrong with UEFI Secure Boot
+----
+
+Can't audit it, signing keys are controlled by vendors, doesn't handle hand off
+ in all cases, depends on possible leaked keys.
+
+Why use Linux instead of vboot2
+----
+
+vboot2 is part of the coreboot tree and is used by Google in the Chromebook
+ system to provide boot time security by verifying the hashes on the coreboot
+ payload. This works well for the specialized Chrome OS on the Chromebook, but
+ is not as flexible as a measured boot solution.
+By moving the verification into the boot scripts we're able to have a much
+ flexible verification system and use more common tools like PGP to sign
+ firmware stages.
+
+What about Trusted GRUB
+----
+
+The mainline grub doesn't have support for TPM and signed kernels, but there is
+ a Trusted grub fork that does. Due to philosophical differences the code might
+ not be merged into the mainline. And due to problems with secure boot (which
+ Trusted Grub builds on), many distributions have signed insecure kernels
+ that bypass all of the protections secure boot promised.
+Additionally, grub is closer to UEFI in that it must have device drivers for
+ all the different boot devices, as well as filesystems. This duplicates the
+ code that exists in the Linux kernel and has its own attack surface.
+
+Using coreboot and Linux as a boot loader allows us to restrict the signature
+ validation to keys that we control. We also have one code base for the device
+ drivers in the Linux-as-a-boot-loader as well as Linux in the operating system.
+
+What is the concern with the Intel Management Engine
+----
+
+"Rootkit in your chipset", "x86 considered harmful", etc
+
+How about the other embedded devices in the system
+----
+
+goodbios, funtenna, etc.
+
+Should we be concerned about the binary blobs
+----
+
+Maybe. x230 has very few (MRC) since it has native vga init.
+
+Why use ancient Thinkpads instead of modern Macbooks
+----
+
+The x230 Thinkpad has coreboot support, TPM, nice keyboards and are very cheap
+ to experiment on. If you're willing to spend a bit more, the Chell Chromebooks
+ (commericaly available as the HP Chromebook 13 G1) has Skylake.
+
+How likely are physical presence attacks vs remote software attacks
+----
+
+Who knows.
+
+Defense in depth vs single layers
+----
+
+Yes.
+
+is it worth doing the hardware modifications
+----
+
+Depends on your threat model.
+
+Should I validate the TPMTOTP on every boot
+----
+
+Probably. I want to make it also do it at S3.
```

Comment:
Trammel wanted to do it. Might refer to issue https://github.com/osresearch/heads/issues/69 which is the closest, unclosed, oldest issue.